File: 1. JPA basic/basic/src/main/java/study/jpa/basic/advancedMapping/code/Team.java
package study.jpa.basic.advancedMapping.code;

import javax.persistence.Entity;
import javax.persistence.GeneratedValue;
import javax.persistence.Id;

/**
 * @author Joonhyuck Hyoung
 */
@Entity
public class Team extends BaseEntity {
    @Id @GeneratedValue
    private Long id;
    private String name;
}

File: 4. Spring Data JPA/springdatajpa/src/main/kotlin/jpabook/springdatajpa/domain/Team.kt
package jpabook.springdatajpa.domain

import net.minidev.json.annotate.JsonIgnore
import javax.persistence.*

/**
 * @author Joonhyuck Hyoung
 */
@Entity
class Team(
    @Id @GeneratedValue(strategy = GenerationType.IDENTITY)
    val id: Long = 0L,
    val name: String,

    @OneToMany(mappedBy = "team")
    val members: MutableList<Member> = mutableListOf()
) {
    companion object {
        fun create(name: String) = Team(name = name)
    }
}
